src/scrape_up/leetcode/leetcode.py

```diff
@@ -0,0 +1,202 @@
+import json
+import requests
+
+class RequestConfig:
+    def __init__(self):
+        self.headers = {}
+
+    def set_headers(self, headers):
+        self.headers = headers
+
+class LeetCode:
+    """
+    ```python
+    leetCode = LeetCode(user="Agastya3636")
+    ```
+    | Methods                                | Details                                                                      |
+    | ---------------------------------------| -----------------------------------------------------------------------------|
+    | `.userContestRanking()`                | Returns rating, global ranking, top percentage, contest count, etc.          |
+    | `.activeBadge()`                       | Returns the active badge of the user.                                        |
+    | `.languageProblemCount()`              | Returns the problems solved by language.                                     |
+    | `.practiceProblemRanking()`            | Returns rank of practice problem by percentage.                              |
+    | `.problemsSolvedByDifficultyNo()`      | Returns problems solved by difficulty count.                                 |
+    """
+
+    def __init__(self, user: str, *, config: RequestConfig = RequestConfig()):
+        self.user = user
+        self.config = config
+        headers = {"User-Agent": "scrapeup"}
+        if self.config.headers == {}:
+            self.config.set_headers(headers)
+        self.user_profile = self.__scrape_user_profile()
+
+    def __scrape_user_profile(self):
+        try:
+            query = f"""
+            query {{
+              matchedUser(username: "{self.user}") {{
+                username
+                languageProblemCount {{
+                  languageName
+                  problemsSolved
+                }}
+                problemsSolvedBeatsStats {{
+                  difficulty
+                  percentage
+                }}
+                submitStatsGlobal {{
+                  acSubmissionNum {{
+                    difficulty
+                    count
+                  }}
+                }}
+                activeBadge {{
+                  displayName
+                  icon
+                }}
+              }}
+              userContestRanking(username: "{self.user}") {{
+                attendedContestsCount
+                rating
+                globalRanking
+                totalParticipants
+                topPercentage
+                badge {{
+                  name
+                }}
+              }}
+            }}
+            """
+            url = 'https://leetcode.com/graphql'
```

Comment:
Hey @agastya3636, you are using the API of the leetcode, even if it is public it is exactly opposite to what we are building.

We are building APIs alternatives, using scraping to overcomes the issues faced while using APIs. Try to add the feature using scraping else I'll not be able to continue with the PR.